packages/trpc/server/routers/viewer/bookings.tsx

```diff
@@ -69,13 +71,29 @@ export const bookingsRouter = createProtectedRouter()
     }),
     async resolve({ ctx, input }) {
       const { bookingId, newLocation: location } = input;
-      const { booking } = ctx;
-
       try {
-        await ctx.prisma.booking.update({
-          where: { id: bookingId },
-          data: { location },
-        });
+
+        const booking = await ctx.prisma.booking.findFirst({
+          where: {
+            id: bookingId
+          },
+          include: {
+            attendees: true,
+            eventType: true,
+            destinationCalendar: true,
+            user: {
+              include: {
+                destinationCalendar: true,
+                credentials: true,
+              }
+            },
+            references: true,
+          }
+        })
```

Comment:
We already do a booking find at [Line 32](https://github.com/calcom/cal.com/pull/3585/files#diff-1f41e2e253ba5e8709f8acc2db09d5ed70727be132f7c63d5ad6af734013cac8L32). This would cause a double query for the same booking, let's try to add the necessary logic to the middleware instead.